lib/components/term.js

```diff
@@ -98,9 +107,43 @@ export default class Term extends Component {
 
     this.getScreenNode().addEventListener('mouseup', this.handleMouseUp);
     this.getScreenNode().addEventListener('mousedown', this.handleMouseDown);
+    this.getScreenNode().addEventListener('contextmenu', this.handleRightClick);
     terms[this.props.uid] = this;
   }
 
+  handleRightClick(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const selection = e.view.getSelection();
+    const {isCollapsed} = selection;
+    const selectedText = selection.toString();
+
+    const selectionMenu = [
+      {
+        type: 'separator'
+      },
+      {
+        label: 'Open Selection as URL',
+        click() {
+          openExternal(selectedText);
+        },
+        enabled: !isCollapsed
+      },
+      {
+        label: 'Search the Web for Selection',
```

Comment:
With Mac OS services you should be able to search with default browser and engine but most likely unusable since Hyper also supports other operative systems ✌️